python/cudf/cudf/core/dataframe.py

```diff
@@ -5426,6 +5436,9 @@ def _prepare_for_rowwise_op(self, method, skipna):
             mask = None
 
         coerced = filtered.astype(common_dtype)
+        if is_pure_dt:
+            # Further convert into cupy friendly types
+            coerced = coerced.astype("int64")
```

Comment:
I know this is our default behavior currently, but we differ from Pandas and may change this in the future:
```suggestion
            coerced = coerced.astype("int64", copy=False)
```